istio-proxy/fetch.sh

```diff
@@ -17,12 +18,48 @@ function set_default_envs() {
     PROXY_GIT_REPO=https://github.com/maistra/proxy
   fi
 
-  if [ -z "${OPENSSL_GIT_BRANCH}" ]; then
-    OPENSSL_GIT_BRANCH=maistra-1.0
+  if [ -z "${PROXY_GIT_BRANCH}" ]; then
+    if [ -z "${PROXY_GIT_COMMIT_HASH}" ]; then
+      PROXY_GIT_COMMIT_HASH=300a927e82a9d952889b8f4de1e0889d4e20e07d
```

Comment:
Does it make sense to use the version in istio-proxy.spec instead?  Regardless, we should probably define all of these commits in there and pass them in as arguments.

Reply:
You can pass them all in and yes, once this is approved I am going to update the spec file.

Reply:
More asking whether it makes sense to update two places every time we change anything.

Reply:
Fetch doesn't have a spec file. rpmbuild does not call fetch. So we can't assume that fetch knows anything about a spec file.

Reply:
No, but update.sh calls spec.

Reply:
OK, once this is merged I will update update.sh to pass the proxy sha (and the other hashes) into fetch.

Reply:
I am debating whether fetch should use the hashes/tarball by default or the branch like it currently does.